### DISCUSSION TITLE
Allow for static map prerendering from versatiles-server

### DISCUSSION THREAD

**Discussion by marcopixel** [versatiles-org/versatiles-documentation#32] [Ideas]
Similar case to https://www.mapbox.com/static-maps, it would be really nice to use static maps as a placeholder while the frontend and tiles are still loading it.

Didn't see it at any other projects and might be really useful too ^^
Love the project and idea behind it ❤️

**Comment by MichaelKreil** [comment]
There is a [server-side renderer using Maplibre Native](https://github.com/maplibre/maplibre-native/tree/main/platform/node). Unfortunately it requires a GPU.
I did some promising experiments with [rendering maps using JavaScript only](https://github.com/versatiles-org/versatiles-renderer). It can draw styled vector data (no sprites or glyphs) as SVG or in a canvas. But it's still at a very early stage.

**Reply by yetzt** [reply]
maplibre native does not require a gpu, but it's fiddly to set up.

**Reply by marcopixel** [reply]
Thanks for the info - seems like the experiment might be the way to go for me 👍

If it's okay for you i would like to keep this discussion open because it might be nice to have this feature built-in in the server implementations ^^

**Reply by MichaelKreil** [reply]
There is also a [Rust prototype for MapLibre Native](https://github.com/maplibre/maplibre-rs). So there might be a package we can integrate into VersaTiles in the near future.

**Reply by yetzt** [reply]
@marcopixel you can compile [maplibre-native](https://github.com/maplibre/maplibre-native/) manually (which can be painful if you don't use the exact os they are supporting) and use the render binary or node bindings to do pretty much what you need. on linux you need to have a virtual framebuffer running (with `xvfb`). 

``` sh
/usr/bin/Xvfb :99 -screen 0 1024x1024x24 &
DISPLAY=:99 EGL_PLATFORM=x11 LD_LIBRARY_PATH="/usr/lib/x86_64-linux-gnu" /path/to/maplibre-native/build/bin/mbgl-render --help
```
